index.html

```diff
@@ -36,4 +37,197 @@ <h1>TODO List</h1>
     </div>
 
 </body>
+
+<!--Retrieve TODO list-->
+<script>
+    const todoInput = document.getElementById('todo-input');
+    const fetchTodosButton = document.getElementById('fetch-todos');
+    const todoList = document.getElementById('todo-list');
+    const todoCount = document.getElementById('todo-count');
+    const addTodoForm = document.getElementById('add-todo-form');
+    const newTodoInput = document.getElementById('new-todo-input');
+    const searchInput = document.getElementById('search-input');
+
+let todos = [];
+
+function renderTodos(todosToRender) {
+    todoList.innerHTML = '';
+
+    // header row
+    const headerRow = document.createElement('li');
+    headerRow.className = 'todo-item header';
+    const header_id = document.createElement('span');
+    header_id.textContent = 'ID';
+    const header_description = document.createElement('span');
+    header_description.textContent = 'TODO Description';
+    const header_userId = document.createElement('span');
+    header_userId.textContent = 'User ID';
+    const header_status = document.createElement('span');
+    header_status.textContent = 'Status';
+    const header_actions = document.createElement('span');
+    header_actions.textContent = 'Actions';
+    headerRow.appendChild(header_id);
+    headerRow.appendChild(header_description);
+    headerRow.appendChild(header_userId);
+    headerRow.appendChild(header_status);
+    headerRow.appendChild(header_actions);
+    todoList.appendChild(headerRow);
+
+ 
+    (todosToRender || todos).forEach((todo, index) => {
```

Comment:
it's better to always pass the `todosToRender `, you can read about pure functions
but this makes the function more predictable and consistent when passing inputs and expecting certain outputs based on them